MergerService/Utils/TaskUtils.cs

```diff
@@ -13,107 +14,154 @@ public class TaskUtils : ITaskUtils
+                {
+                    this._logger.LogWarning(e, "Error deserializing returned task");
+                    return null;
+                }
             }
+
+            // TODO: add heartbeat stop method
         }
 
         public void NotifyOnCompletion(string jobId, string taskId)
         {
-            // Notify overseer on task completion
-            string baseUrl = this._configuration.GetConfiguration("TASK", "overseerUrl");
-            string relativeUri = $"tasks/{jobId}/{taskId}/completed";
-            string url = new Uri(new Uri(baseUrl), relativeUri).ToString();
-            _ = this._httpClient.PostDataString(url, null, false);
+            using (var activity = this._activitySource.StartActivity("notify overseer on task completion"))
+            {
+                // Notify overseer on task completion
+                this._logger.LogInformation($"Notifying overseer on completion, job: {jobId}, task: {taskId}");
+                string relativeUri = $"tasks/{jobId}/{taskId}/completed";
+                string url = new Uri(new Uri(_overseerUrl), relativeUri).ToString();
+                _ = this._httpClient.PostDataString(url, null);
+            }
         }
 
-        private void Update(string jobId, string taskId, FormUrlEncodedContent content)
+        private void Update(string jobId, string taskId, HttpContent content)
         {
-            string baseUrl = this._configuration.GetConfiguration("TASK", "jobManagerUrl");
             string relativeUri = $"jobs/{jobId}/tasks/{taskId}";
-            string url = new Uri(new Uri(baseUrl), relativeUri).ToString();
-            _ = this._httpClient.PutDataString(url, content, false);
+            string url = new Uri(new Uri(_jobManagerUrl), relativeUri).ToString();
+            _ = this._httpClient.PutDataString(url, content);
         }
 
         public void UpdateProgress(string jobId, string taskId, int progress)
         {
-            var content = new FormUrlEncodedContent(new[]
+            using (var activity = this._activitySource.StartActivity("update task progress"))
             {
-                new KeyValuePair<string, string>("percentage", progress.ToString())
-            });
-            Update(jobId, taskId, content);
+                // activity.AddTag("progress", progress);
+
+                using (var content = new StringContent(JsonConvert.SerializeObject(new
+                {
+                    percentage = progress
+                }, this._jsonSerializerSettings)))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                    Update(jobId, taskId, content);
+                }
+            }
         }
 
         public void UpdateCompletion(string jobId, string taskId)
         {
-            var content = new FormUrlEncodedContent(new[]
+            using (var activity = this._activitySource.StartActivity("update task completed"))
```

Comment:
save activity into variable only if you there metadata should be add to the trace,
otherwise just use:
using (this._activitySource.StartActivity("dequeue task"))

Reply:
Removed.